### PR TITLE
Add Openstack driver

### PR DESCRIPTION
**Pull request by dynamiccast**
Allow Nanocloud to connect to an openstack to boot up new execution servers.

### CODE REVIEW

File: api/drivers/openstack/driver.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+              this._client.allocateNewFloatingIp((err, ips) => {
+                if (err) {
+                  return reject(err);
+                } else {
+                  this._client.addFloatingIp(server, ips.ip, (err) => {
+                    if (err) {
+                      return reject(err);
+                    }
+                  });
+                  return resolve(Machine.create({
```
Comment (willrnch) [inline api/drivers/openstack/driver.js:156]: `Machine.create` returns a promise. No need to pass it to `resolve`

File: api/drivers/openstack/driver.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        let password;
+        if (config.openstackMachinePassword === '') {
+          password = randomstring.generate(12) + 'a8A';
+        } else {
+          password = config.openstackMachinePassword;
+        }
+
+        let userData = new Buffer(`#ps1_sysnative
+        REG.exe Add HKLM\\Software\\Microsoft\\ServerManager /V DoNotOpenServerManagerAtLogon /t REG_DWORD /D 0x1 /F
+        net user ${config.openstackMachineUsername} ${password}
```
Comment (willrnch) [inline api/drivers/openstack/driver.js:112]: What if `config. openstackMachineUsername` or `password` contain characters interpreted by powershell?

File: api/drivers/openstack/driver.js (hunk 1 above)
Comment (Leblantoine) [inline api/drivers/openstack/driver.js:156]: It does not work without `resolve`

File: api/drivers/openstack/driver.js (hunk 2 above)
Comment (Leblantoine) [inline api/drivers/openstack/driver.js:112]: All characters are available now except the double quote, see #69

**Comment by Leblantoine** [comment]
Closed in favor of rebase on master PR : #70

**Pull request closed without merging**